### PR TITLE
Improved persistence

### PR DESCRIPTION
**Pull request by kickroot**
Removed all AWS specific persistence fields and move to MagpieResource, allowing us to persist other cloud providers.

### CODE REVIEW

File: magpie-aws/src/test/java/io/openraven/magpie/plugins/aws/discovery/services/IAMRoleDiscoveryIT.java [hunk 1]
```diff
@@ -67,19 +67,19 @@ public void testRoleDiscovery() {
...
+    assertEquals(ACCOUNT, contents.get("accountId").asText());
+    assertEquals(BASE_REGION.toString(), contents.get("region").asText());
   }
 
   private void assertConfiguration(MagpieEnvelope envelope) {
     var configuration = envelope.getContents().get("configuration");
     assertEquals("/", configuration.get("path").asText());
     assertNotNull(configuration.get("roleId"));
-    assertTrue(configuration.get("arn").asText().contains(ROLE_NAME));
+    assertTrue(configuration.get("assetId").asText().contains(ROLE_NAME));
```
Comment (stefansjfw) [inline magpie-aws/src/test/java/io/openraven/magpie/plugins/aws/discovery/services/IAMRoleDiscoveryIT.java:82]: this should remain `arn`. This is AWS specific configuration data received from AWS client

File: magpie-aws/src/test/java/io/openraven/magpie/plugins/aws/discovery/services/IAMGroupDiscoveryIT.java [hunk 2]
```diff
@@ -94,7 +94,7 @@ private void assertAttachedPolicies(MagpieEnvelope envelope) {
 
     var policy = attachedPolicies.get(0);
     assertEquals("managedDataAccess", policy.get("name").asText());
-    assertEquals(String.format("arn:aws:iam::000000000000:policy/%s", MANAGED_POLICY_NAME), policy.get("arn").asText());
+    assertEquals(String.format("arn:aws:iam::000000000000:policy/%s", MANAGED_POLICY_NAME), policy.get("assetId").asText());
```
Comment (stefansjfw) [inline magpie-aws/src/test/java/io/openraven/magpie/plugins/aws/discovery/services/IAMGroupDiscoveryIT.java:97]: same here. should be `arn`

**Comment by stefansjfw** [comment]
I took the liberty of fixing failing tests and resolving conflicts :)

**Review by PrzemyslawTusinski — APPROVED** [review]

**Comment by stefansjfw** [comment]
CI is still failing because of PolicyDiscovery test as we switched from discovering only LOCAL policies to all policies. I'm working on fixing this one too

**Comment by stefansjfw** [comment]
Build and tests should be green now

File: magpie-aws/src/test/java/io/openraven/magpie/plugins/aws/discovery/services/IAMAccountDiscoveryIT.java [hunk 3]
```diff
@@ -58,17 +59,16 @@ private void assertConfiguration(MagpieEnvelope envelope) {
     var passwordPolicy = supplementaryConfiguration.get("PasswordPolicy");
     assertEquals("{\"minimumPasswordLength\":8,\"requireSymbols\":true,\"requireNumbers\":true,\"requireUppercaseCharacters\":true,\"requireLowercaseCharacters\":true,\"allowUsersToChangePassword\":true,\"expirePasswords\":true,\"maxPasswordAge\":200,\"passwordReusePrevention\":1,\"hardExpiry\":true}",
       passwordPolicy.toString());
-    assertNotNull(supplementaryConfiguration.get("summaryMap").toString());
```
Comment (stefansjfw) [inline magpie-aws/src/test/java/io/openraven/magpie/plugins/aws/discovery/services/IAMAccountDiscoveryIT.java:61]: We removed accountSummary discovery here https://github.com/openraven/magpie/pull/115

Don't remember what was the reason

Comment (PrzemyslawTusinski) [reply magpie-aws/src/test/java/io/openraven/magpie/plugins/aws/discovery/services/IAMAccountDiscoveryIT.java:61]: It wasn't removed it was moved from supplementaryConfiguration to configuration

**Comment by belosh59** [comment]
@stefansjfw  thanks for resolving test failures. Overall LGTM

**Review by belosh59 — APPROVED** [review]

File: magpie-aws/src/test/java/io/openraven/magpie/plugins/aws/discovery/services/IAMAccountDiscoveryIT.java (hunk 3 above)
Comment (stefansjfw) [reply magpie-aws/src/test/java/io/openraven/magpie/plugins/aws/discovery/services/IAMAccountDiscoveryIT.java:61]: Ah, I see it now. Thanks! We lost it somewhere during transition to `MagpieResource`. Pushed fix

**Pull request merged**